torch/distributed/_composable/fsdp/_fsdp_state.py

```diff
@@ -13,13 +14,16 @@
 )
 from torch.distributed.utils import _to_kwargs
 from torch.utils._pytree import tree_flatten, tree_map
+
 from ._fsdp_api import MixedPrecisionPolicy
 from ._fsdp_common import _cast_fp_tensor, TrainingState
 from ._fsdp_param_group import FSDPCommContext, FSDPParamGroup
 
 if TYPE_CHECKING:
     from ._fsdp_param import FSDPParam
 
+logger = logging.getLogger(__name__)
```

Comment:
indirect comment but want to understand how log level works

Is the log level controled by `__name__` and whether it matches  `register_log("fsdp", ["torch.distributed._composable.fsdp"])` ?

Reply:
the log level is controlled by the user at runtime with TORCH_LOGS or `torch._logging.set_logs`. The `__name__` is used to ensure messages are routed to the correct location. The logging system is hierarchical, so if you setup logging for a module A, any submodules A.B, or A.C.D, will be handled by the infra for A. In this case, we setup logging for `torch.distributed._composable.fsdp` so any submodules will route through the handling of this module.

Reply:
thanks for explaining torch_logs. I get it now.

Probably unit test is the immediate blocker then

Reply:
switched to `torch.distributed._composable.fsdp`